Lib/http/client.py

```diff
@@ -1086,6 +1090,7 @@ def putrequest(self, method, url, skip_host=False,
             raise CannotSendRequest(self.__state)
 
         # Save the method for use later in the response phase
+        self._validate_method(method)
```

Comment:
Placing the validation right behind this comment is somewhat misleading -- the comment explains the assignment, but the validation is *not* just for the assignment -- its primary use is for the construction of the `request` variable a few lines nelow (L1099 in this version). I recommend moving this up and placing blank lines around it.